src/builder.ts

```diff
@@ -135,3 +135,37 @@ export class CommandBuilder {
 
 export const command = (executable: string, ...args: string[]) =>
   new CommandBuilder({ type: 'spawn', executable, args })
+
+export const script = (template: TemplateStringsArray, ...args: unknown[]) => {
+  const source = template
+    .map((s: string, i: number) => {
+      const arg = args[i]
+
+      if (arg === null || arg === undefined) {
+        return s
+      }
+      else if (typeof arg === 'string') {
+        return `${s}${quote(arg)}`
+      }
+      else if (typeof arg === 'number' || typeof arg === 'boolean') {
+        return `${s}${arg}`
+      }
+      else {
+        throw new TypeError(
+          `Unexpected template argument of type ${typeof arg}`
+        )
+      }
+    })
+    .join('')
+
+  const commands = source
+    .split(';')
+    .flatMap(s => s.split('\n'))
+    .map(s => s.trim())
+    .filter(s => s.length > 0)
```

Comment:
Once again, good catch, I decided to do this instead:

```typescript
export const script = {
  posix: (template: TemplateStringsArray, ...args: unknown[]) => {
    const src = makeSource(template, args)
    return command('sh', '-').writeStdin(Buffer.from(src))
  },
  powershell: (template: TemplateStringsArray, ...args: unknown[]) => {
    const src = makeSource(template, args)
    return command('PowerShell', '-Command', '-').writeStdin(Buffer.from(src))
  },
}
```

Which can be used like this:

```typescript
await script.posix`
  ls
  echo hello
  echo world
`.exec()

await script.powershell`
  dir
  echo hello
  echo world
`.exec()
```

I decided to not support `cmd.exe` since it does not support reading a script from stdin. The closest we have is `cmd.exe /C "a single command"` which isn't very useful.

Thank you for your insight 🙂

Reply:
NB: `script.posix` should work fine with Git Bash on Windows as well.